tracing-subscriber/src/subscribe.rs

```diff
@@ -1072,10 +1072,10 @@ where
     where
         C: for<'lookup> LookupSpan<'lookup>,
     {
-        let subscriber = self.subscriber.as_ref()?;
-        let current = subscriber.current_span();
+        let collector = self.collector.as_ref()?;
+        let current = collector.current_span();
         let id = current.id()?;
-        let span = subscriber.span(&id);
+        let span = collector.span(&id);
         debug_assert!(
             span.is_some(),
             "the subscriber should have data for the current span ({:?})!",
```

Comment:
we should probably also change the panic message:
```suggestion
            "the collector should have data for the current span ({:?})!",
```